RecoLocalCalo/EcalRecAlgos/src/PulseChiSqSNNLS.cc

```diff
@@ -410,8 +411,8 @@ bool PulseChiSqSNNLS::NNLS() {
     
     //adaptive convergence threshold to avoid infinite loops but still
     //ensure best value is used
-    if (iter%50==0) {
-      threshold *= 10.;
+    if (iter % 16 == 0) {
+      threshold *= 2;
```

Comment:
This adaptive convergence business was added as part of the mitigation to pathological events which caused infinite loops in the HLT at some point.  (This may have also been conditions-dependent)